x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/page.tsx

```diff
@@ -39,6 +42,22 @@ export const Page: FC = () => {
 
   const location = useLocation();
   const selectedTabId = useMemo(() => location.pathname.split('/').pop(), [location]);
+  const mapJobId = useMemo(() => {
+    const { _g }: Record<string, any> = parse(location.search, { sort: false });
+    let jobId: string | undefined;
+
+    if (_g !== undefined) {
+      let globalState: any = null;
+      try {
+        globalState = decode(_g);
+      } catch (error) {
+        // eslint-disable-next-line no-console
+        console.error('Could not parse global state');
+      }
+      jobId = globalState.ml?.jobId;
+    }
+    return jobId;
+  }, [location]);
```

Comment:
the URL state should take care of it. check https://github.com/elastic/kibana/blob/f4534674f2dcdf8cc0b0c910142e5bda28048d1c/x-pack/plugins/ml/public/application/util/url_state.tsx